include/cuda/functional

```diff
@@ -71,23 +71,20 @@ class __return_type_wrapper {
  public:
   __return_type_wrapper() = delete;
 
-  _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 
-  __return_type_wrapper(__return_type_wrapper const&) = default;
-
```

Comment:
@senior-zero do you remember why you added that defaulted copy constructor?

Reply:
https://github.com/NVIDIA/libcudacxx/pull/386

Reply:
Nope, that defaulted copy constructor was an oversight on my part. There are no base classes and no move constructors / assignments defined, so defaulting the copy constructor has no effect.

The previous issue was that the templatized constructor did also match the signnature of the copy constructor creating an ambiguity. With the added constraint the templated constructor does not interfere anymore and we do not need the default the copy constructor.

Reply:
What added constraint are we talking about?

Reply:
#386 changed this:
```cpp
template <class _Fn>
  _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 _LIBCUDACXX_INLINE_VISIBILITY
  explicit __return_type_wrapper(_Fn&& __fn) noexcept
      : __fn_(_CUDA_VSTD::forward<_Fn>(__fn)) {}
```
to this
```cpp
  template <class _Fn,
            class = _CUDA_VSTD::__enable_if_t<_CUDA_VSTD::is_same<_CUDA_VSTD::__decay_t<_Fn>, _DecayFn>::value>>
  _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 _LIBCUDACXX_INLINE_VISIBILITY 
  explicit __return_type_wrapper(_Fn &&__fn) noexcept
    : __fn_(_CUDA_VSTD::forward<_Fn>(__fn)) {}
```

Reply:
Oh, I was confused because #386 had added the constraint and the default ctor, so I assumed it was necessary.